src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractInProcLanguageClient.cs

```diff
@@ -176,5 +186,78 @@ public Task OnServerInitializeFailedAsync(Exception e)
             // We don't need to provide additional exception handling here, liveshare already handles failure cases for this server.
             return Task.CompletedTask;
         }
+
+        internal static async Task<InProcLanguageServer> CreateAsync(
```

Comment:
this was entirely copied from InProcLanguageServer and moved here (since we need to initialize the jsonrpc connection outside of it).